pkg/util/v1alpha2/metricscollector/metricscollector.go

```diff
@@ -0,0 +1,116 @@
+package metricscollector
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
```

Comment:
Can we use controller runtime client itself instead of directly using client-go?

Reply:
This is only used to get the in-cluster config, which needs the REST client. The controller runtime client also calls rest client under the hood: https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/client/config/config.go#L78

Reply:
Changed to controller-runtime